main.go

```diff
@@ -0,0 +1,205 @@
+	"github.com/testground/sdk-go/runtime"
+	"github.com/testground/sdk-go/sync"
+
+	svrcmd "github.com/cosmos/cosmos-sdk/server/cmd"
+)
+
+var testcases = map[string]interface{}{
+	"capp-1": run.InitializedTestCaseFn(runSync),
+}
+
+func main() {
+	run.InvokeMap(testcases)
+}
+
+type ValidatorNode struct {
+	PubKey string
+	IP     net.IP
+}
+
+func AddPersistentPeers(path string, peers []string) error {
+
+	var peersStr bytes.Buffer
+	var port int = 26656
+	var separator string = ","
+	for k, peer := range peers {
+		if k == (len(peers) - 1) {
+			separator = ""
+		}
+		peersStr.WriteString(fmt.Sprintf("%s:%d%s", peer, port, separator))
+	}
+
+	fh, err := os.OpenFile(path, os.O_RDWR, 0777)
+	if err != nil {
+		return err
+	}
+
+	viper.SetConfigType("toml")
+	err = viper.ReadConfig(fh)
+	if err != nil {
+		return err
+	}
+
+	viper.Set("p2p.persistent-peers", peersStr.String())
+	err = viper.WriteConfigAs(path)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func runSync(runenv *runtime.RunEnv, initCtx *run.InitContext) error {
+	// const home string = "/Users/bidon4/.celestia-app-1"
+	ctx := context.Background()
+	client := initCtx.SyncClient
+	netclient := network.NewClient(client, runenv)
+
+	netclient.MustWaitNetworkInitialized(ctx)
+
+	config := network.Config{
+		// Control the "default" network. At the moment, this is the only network.
+		Network: "default",
+
+		// Enable this network. Setting this to false will disconnect this test
+		// instance from this network. You probably don't want to do that.
+		Enable: true,
+
+		// Set the traffic shaping characteristics.
+		Default: network.LinkShape{
+			Latency:   100 * time.Millisecond,
+			Bandwidth: 1 << 20, // 1Mib
+		},
+
+		// Set what state the sidecar should signal back to you when it's done.
+		CallbackState: "network-configured",
+	}
+
+	topic := sync.NewTopic("ip-allocation", "")
+	seq := client.MustPublish(ctx, topic, "")
+
+	config.IPv4 = runenv.TestSubnet
+	// Use the sequence number to fill in the last two octets.
+	//
+	// NOTE: Be careful not to modify the IP from `runenv.TestSubnet`.
+	// That could trigger undefined behavior.
+	ipC := byte((seq >> 8) + 1)
+	ipD := byte(seq)
+	config.IPv4.IP = append(config.IPv4.IP[0:2:2], ipC, ipD)
+
+	err := netclient.ConfigureNetwork(ctx, &config)
+	if err != nil {
+		runenv.RecordCrash(err)
+		return err
+	}
+
+	var home string
+	switch seq {
+	case 1:
+		home = runenv.StringParam("app1")
+	case 2:
+		home = runenv.StringParam("app2")
+	case 3:
+		home = runenv.StringParam("app3")
+	case 4:
+		home = runenv.StringParam("app4")
+	case 5:
+		home = runenv.StringParam("app5")
+	case 6:
+		home = runenv.StringParam("app6")
+	case 7:
+		home = runenv.StringParam("app7")
+	case 8:
+		home = runenv.StringParam("app8")
+	case 9:
+		home = runenv.StringParam("app9")
+	case 10:
+		home = runenv.StringParam("app10")
+	}
+
+	// if seq == 1 {
+	// 	home = runenv.StringParam("app1")
+	// } else {
+	// 	home = runenv.StringParam("app2")
+	// }
```

Comment:
home = runenv.StringParam(fmt.Sprintf("app%d", seq) ?

Reply:
I was playing aroound and forgot to do. Thanks for pointing this out! Changed